### PR TITLE
Extended installation settings

### PR DESCRIPTION
**Pull request by pnowy**
I've added two installation settings which would be good to see from my project perspective:

1. Custom installation directory (added `withInstallationDirectory(File)` method for builder)
2. Option to disable cleaning data directory after each server stop (added `withCleanInstallationDirectoryOnStop(boolean)` method for builder)

The default behavior has not been touched. Any comments are welcome.

### CODE REVIEW

File: test-base/src/main/groovy/pl/allegro/tech/embeddedelasticsearch/EmbeddedElasticConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package pl.allegro.tech.embeddedelasticsearch;
+
+/**
+ * Common test configuration.
+ */
+public interface EmbeddedElasticConfiguration {
```
Comment (gaczm) [inline test-base/src/main/groovy/pl/allegro/tech/embeddedelasticsearch/EmbeddedElasticConfiguration.java]: please move esJavaOpts here as well

File: es50-test/src/test/groovy/pl/allegro/tech/embeddedelasticsearch/CustomInstallationDirectorySpec.groovy [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        INSTALLATION_DIRECTORY.deleteDir()
+    }
+
+    def cleanupSpec() {
+        INSTALLATION_DIRECTORY.deleteDir()
+    }
+    
+    def "should install embedded elastic on custom directory"() {
+        when:
+        embeddedElastic.start()
```
Comment (gaczm) [inline es50-test/src/test/groovy/pl/allegro/tech/embeddedelasticsearch/CustomInstallationDirectorySpec.groovy]: In all other tests we have indentation after given,when,then labels. Please, apply that style here too

File: test-base/src/main/groovy/pl/allegro/tech/embeddedelasticsearch/EmbeddedElasticConfiguration.java (hunk 1 above)
Comment (pnowy) [inline test-base/src/main/groovy/pl/allegro/tech/embeddedelasticsearch/EmbeddedElasticConfiguration.java]: Sure

File: es50-test/src/test/groovy/pl/allegro/tech/embeddedelasticsearch/CustomInstallationDirectorySpec.groovy (hunk 2 above)
Comment (pnowy) [inline es50-test/src/test/groovy/pl/allegro/tech/embeddedelasticsearch/CustomInstallationDirectorySpec.groovy]: Done, I've missed it ;)

**Pull request merged**